### DISCUSSION TITLE
Is it possible to execute an external program?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#948] [Ask a question] [answered]
Hello! 
I'm playing around with [ImageMagic](https://imagemagick.org/index.php) and Yellow. I'm trying to use filters and use, as starting point the following function as a test inside the `image.php`:

```
    public function nashville($image)
    {
        $image = $this->execute("convert $image -contrast -modulate 100,150,100 -auto-gamma $image");
        return $image;
    }
```
Unfortunately I get an error message. 

`PHP Fatal error:  Uncaught Error: Call to undefined method YellowImage::execute() in /var/www/html/test/system/extensions/image.php:265\nStack trace:\n#0 ...`

Could it be that `execute()` is not allowed in Yellow? Perhaps for safety reasons? If it is allowed, maybe someone has a tip on how I can proceed? &#128543;

**Comment by schulle4u** [comment, marked as answer]
Yellow doesn't block any PHP functions. Whatever your web host allows, it is also allowed in Yellow extensions. The error message states that you are trying to call an undefined method in your code. Maybe try [exec](https://www.php.net/manual/en/function.exec.php) instead?

**Comment by PetersOtto** [comment]
Hi Steffen,
thank you very much for your answer. Then I can rule out Yellow as a source of error. 
I think I'll have to get to grips with ImageMagick a bit more first.
Have a nice weekend!

**Comment by GiovanniSalmeri** [comment]
Please remark that:

1. `exec` and [similar functions](https://www.php.net/manual/en/book.exec.php) can be disabled in a specific installation of PHP for security reasons;
2. Imagick may be installed as an [extension of PHP](https://www.php.net/manual/en/book.imagick.php): in this case it is much better to use the corresponding functions of PHP rather than execute Imagick as an external program.

**Comment by PetersOtto** [comment]
Hello Giovanni.
Thanks for the tip!
